### PR TITLE
feat(common): also verify traefik, prometheus and cnpg automatically

### PR DESCRIPTION
**Pull request by Ornias1993**
**Description**
With the verificator mostly polished, we still need to implement non-"additional" operators.
As well as adding traefik to be essentially treated as an operator when it is the primary install, due to CRDs

Also changes the way how ingressclass middleware names are generated, which should fix some of the ongoing middleware issues when users use ingressclass

**⚙️ Type of change**

- [x] ⚙️ Feature/App addition
- [x] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**

**📃 Notes:**
traefik middleware naming changes should be backwards compatible to previous version (pointing to `tc-system` namespace)

**✔️ Checklist:**

- [ ] ⚖️ My code follows the style guidelines of this project
- [ ] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [ ] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [ ] 🪞 I have opened a PR on [truecharts/containers](https://github.com/truecharts/containers) adding the container to TrueCharts mirror repo.
- [ ] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

### CODE REVIEW

File: library/common/templates/lib/util/_chartcontext.tpl [hunk 1]
```diff
@@ -42,7 +42,18 @@
...
+
       {{- $namespace := "tc-system" -}}
+      {{- if $.Values.operator.traefik -}}
+        {{- if $.Values.operator.traefik.namespace -}}
+          {{- $namespace := $.Values.operator.traefik.namespace -}}
+        {{- end -}}
+      {{- end -}}
+      {{- if $selectedIngress.ingressClassName }}
+        {{- $namespace = $selectedIngress.ingressClassName -}}
+      {{- end }}
```
Comment (stavros-k) [inline library/common/templates/lib/util/_chartcontext.tpl:55]:
```suggestion
      {{- end -}}
```

File: library/common/templates/lib/util/_register_operator.tpl [hunk 2]
```diff
@@ -1,4 +1,7 @@
 {{- define "tc.v1.common.lib.util.operator.register" -}}
+  {{- if .Values.crds.install -}}
+    {{- $_ := set .Values.operator register true -}}
```
Comment (stavros-k) [inline library/common/templates/lib/util/_register_operator.tpl:3]:
```suggestion
    {{- $_ := set .Values.operator "register" true -}}
```

File: library/common/templates/lib/util/_verify_operator.tpl [hunk 3]
```diff
@@ -1,7 +1,39 @@
...
+    {{- $operatorList := .Values.operator.verify.additionalOperators -}}
+
+    {{- $cnpg := false -}}
+    {{- range $opName := .Values.cnpg -}}
+      {{- if .enabled -}}
+        {{- $cnpg := true -}}
+      {{- end -}}
+    {{- end -}}
+    {{- if $cnpg -}}
+      {{- $operatorList = append $operatorList "cloudnative-pg" -}}
```
Comment (stavros-k) [inline library/common/templates/lib/util/_verify_operator.tpl:13]:
```suggestion
      {{- $operatorList = mustAppend $operatorList "cloudnative-pg" -}}
```

File: library/common/templates/lib/util/_verify_operator.tpl [hunk 4]
```diff
@@ -1,7 +1,39 @@
...
+    {{- end -}}
+
+    {{- $ingress := false -}}
+    {{- range $opName := .Values.ingress -}}
+      {{- if .enabled -}}
+        {{- $ingress := true -}}
+      {{- end -}}
+    {{- end -}}
+    {{- if $ingress -}}
+      {{- $operatorList = append $operatorList "traefik" -}}
```
Comment (stavros-k) [inline library/common/templates/lib/util/_verify_operator.tpl:23]:
```suggestion
      {{- $operatorList = mustAppend $operatorList "traefik" -}}
```

File: library/common/templates/lib/util/_verify_operator.tpl [hunk 5]
```diff
@@ -1,7 +1,39 @@
...
+    {{- end -}}
+
+    {{- $metrics := false -}}
+    {{- range $opName := .Values.metrics -}}
+      {{- if .enabled -}}
+        {{- $metrics := true -}}
+      {{- end -}}
+    {{- end -}}
+    {{- if $metrics -}}
+      {{- $operatorList = append $operatorList "prometheus" -}}
```
Comment (stavros-k) [inline library/common/templates/lib/util/_verify_operator.tpl:33]:
```suggestion
      {{- $operatorList = mustAppend $operatorList "prometheus" -}}
```

File: library/common/templates/spawner/_portal.tpl [hunk 6]
```diff
@@ -70,11 +70,19 @@
...
+        {{- $traefikNamespace := "tc-system" -}}
+        {{- if $.Values.operator.traefik -}}
+          {{- if $.Values.operator.traefik.namespace -}}
+            {{- $traefikNamespace := $.Values.operator.traefik.namespace -}}
+          {{- end -}}
+        {{- end -}}
         {{- if $selectedIngress.ingressClassName }}
-        {{- $hook = ( printf "portalhook-%s" $selectedIngress.ingressClassName ) -}}
+          {{- $traefikNamespace = (printf "ix-%s" $selectedIngress.ingressClassName) -}}
         {{- end }}
```
Comment (stavros-k) [inline library/common/templates/spawner/_portal.tpl:83]:
```suggestion
        {{- end -}}
```

File: library/common/templates/lib/util/_register_operator.tpl (hunk 2 above)
Comment (stavros-k) [reply library/common/templates/lib/util/_register_operator.tpl:3]: This is the only "bug", rest are just styling!

**Review by stavros-k — COMMENTED** [review]
Did a quick pass (found a typo in the `set` func), didn't check logic.

File: library/common/values.yaml [hunk 7]
```diff
@@ -165,7 +165,7 @@ workload:
               port: "{{ $.Values.service.main.ports.main.targetPort | default .Values.service.main.ports.main.port }}"
             startup:
               enabled: true
-              type: "tcp"
+              type: "{{ .Values.service.main.ports.main.protocol }}"
```
Comment (Ornias1993) [inline library/common/values.yaml:168]:
```suggestion
              type: "tcp"
```

File: library/common/values.yaml [hunk 8]
```diff
@@ -328,7 +328,7 @@ netshootImage:
 
 tailscaleImage:
   repository: tccr.io/truecharts/tailscale
-  tag: v1.42.0@sha256:d29d3f96acd24c14c231da4d28de5f77744d605151c39609d5bdf4245a157c39
+  tag: v1.40.1@sha256:013d7c85ca3e5eb4afc3a635dc09d1677a701fa9daf250ffe2d15a00b0583418
```
Comment (Ornias1993) [inline library/common/values.yaml:331]:
```suggestion
  tag: v1.42.0@sha256:d29d3f96acd24c14c231da4d28de5f77744d605151c39609d5bdf4245a157c39
```

File: library/common/values.yaml [hunk 9]
```diff
@@ -810,6 +822,3 @@ solr:
   creds: {}
   manifestManager:
     enabled: false
```
Comment (Ornias1993) [inline library/common/values.yaml:824]:
```suggestion
    enabled: false

# -- List of extra objects to deploy with the release
extraTpl: []

```

File: library/common/values.yaml [hunk 10]
```diff
@@ -813,3 +825,4 @@ solr:
 
 # -- List of extra objects to deploy with the release
 extraTpl: []
+
```
Comment (Ornias1993) [inline library/common/values.yaml:828]:
```suggestion
```

**Pull request merged**